src/rpc/core_rpc_server.cpp

```diff
@@ -2781,6 +2782,142 @@ namespace cryptonote
+
+    crypto::hash hash;
+
+    uint64_t height = 0;
+
+    for (auto i = 0u; i < NUM_ITERATIONS; ++i)
+    {
+      /// incorporate seed so that the results couldn't be cached;
+      /// increment seed to break cycles;
+      height = height ^ (seed++);
+      height = height % (max_height + 1);
+
+      hash = core.get_block_id_by_height(height);
+
+      using blob_t = cryptonote::blobdata;
+      using block_pair_t = std::pair<blob_t, block>;
+
+      /// pick a random byte from the block blob
+      std::vector<block_pair_t> blocks;
+      std::vector<blob_t> txs;
+      if (!core.get_blockchain_storage().get_blocks(height, 1, blocks, txs)) {
+        MERROR("Could not query block at requested height: " << height);
+        return 0;
+      }
+      const blob_t &blob = blocks.at(0).first;
+      const uint64_t byte_idx = service_nodes::uniform_distribution_portable(mt, blob.size());
+      uint8_t byte = blob[byte_idx];
+
+      /// pick a random byte from a random transaction blob if found
+      if (!txs.empty()) {
+        const uint64_t tx_idx = service_nodes::uniform_distribution_portable(mt, txs.size());
+        const blob_t &tx_blob = txs[tx_idx];
+
+        /// not sure if this can be empty, so check to be safe
+        if (!tx_blob.empty()) {
+          const uint64_t byte_idx = service_nodes::uniform_distribution_portable(mt, tx_blob.size());
+          const uint8_t tx_byte = tx_blob[byte_idx];
+          byte ^= tx_byte;
+        }
+
+      }
+
+      {
+        /// reduce hash down to 8 bytes
+        const uint64_t n1 = *reinterpret_cast<uint64_t*>(hash.data);
+        const uint64_t n2 = *reinterpret_cast<uint64_t*>(hash.data + 8);
+        const uint64_t n3 = *reinterpret_cast<uint64_t*>(hash.data + 16);
+        const uint64_t n4 = *reinterpret_cast<uint64_t*>(hash.data + 24);
+
+        /// Note that byte (obviously) only affects the lower byte
+        /// of height, but that should be sufficient in this case
+        height = n1 ^ n2 ^ n3 ^ n4 ^ byte;
+      }
+
+    }
+
+    return height;
+  }
+
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_perform_blockchain_test(const COMMAND_RPC_PERFORM_BLOCKCHAIN_TEST::request& req,
+                                                   COMMAND_RPC_PERFORM_BLOCKCHAIN_TEST::response& res,
+                                                   epee::json_rpc::error& error_resp,
+                                                   const connection_context* ctx)
+  {
+    PERF_TIMER(on_perform_blockchain_test);
+
+    /// 1. Check signature
+    {
+      crypto::public_key my_pubkey;
+      crypto::secret_key my_seckey;
+      if (!m_core.get_service_node_keys(my_pubkey, my_seckey)) {
+        error_resp.code    = CORE_RPC_ERROR_CODE_WRONG_PARAM;
+        error_resp.message = "Daemon has not been started in service node mode, please relaunch with --service-node flag.";
```

Comment:
Has been brought up before, I agree there's not much point to it since everyone syncs the service node list, so they can just check that list if it is dependent on being registered. At the time it was decided against since it's a relatively minor step in comparison to getting the whole service node up and running.

We can default it on in the next release since it actually simplifies the UX.